src/parser/postgresparser.cpp

```diff
@@ -581,10 +582,21 @@ std::unique_ptr<AbstractExpression> PostgresParser::ColumnRefTransform(ParseResu
       if (all_columns)
         result = std::make_unique<TableStarExpression>(table_name);
       else if (alias != nullptr)
+        /*
```

Comment:
Thanks for this comment here, I was pretty confused above where we were using `reinterpret_cast` to go from an expression pointer to an `alias_oid_t`.

Reply:
Thanks, but I don't think this comment actually addresses that. This comment is explaining that for SELECT queries at some point we'll have to add a serial number to the `ColumnValueExpression`'s table `AliasType`, so that it matches the serial number of the corresponding `AliasType` that belongs to the `TableRef`.

Later in the binder `ColumnValueExpression`s are matched to `TableRef`s using the `AliasType` so they need to be equivalent.

On a separate note, we are using `reinterpret_cast` to go from an expression ponter to an `alias_oid_t` to create the serial number for the `ColumnValueExpression`'s column `AliasType`. The assumption here is that the pointer will be unique. That was already there though so I didn't really add a comment for it. Though I can now.